modules/qna/src/views/FormModal.jsx

```diff
@@ -88,87 +88,80 @@ export default class FormModal extends Component {
       redirectFlow: isRedirect && !item.redirectFlow,
       redirectNode: isRedirect && !item.redirectNode
     }
+    const hasDuplicates = this.isQuestionDuplicated()
 
-    this.setState({ invalidFields })
-    return some(invalidFields)
-  }
-
-  trimItemQuestions = questions => {
-    return questions.map(q => q.trim()).filter(q => q !== '')
+    this.setState({ invalidFields, hasDuplicates, errorMessage: undefined })
+    return some(invalidFields) || hasDuplicates
   }
 
-  onCreate = event => {
-    event.preventDefault()
-    if (this.validateForm()) {
-      this.setState({ isValidForm: false })
-
-      return
-    }
-
-    if (!this.state.isValidForm) {
-      this.setState({ isValidForm: true })
-    }
-
+  isQuestionDuplicated() {
     const item = this.state.item
     const questions = this.trimItemQuestions(item.questions)
 
-    return this.props.bp.axios.post('/mod/qna/questions', { ...item, questions }).then(() => {
-      this.props.fetchData()
-      this.closeAndClear()
-    })
+    const dupes = _.filter(questions, (value, index) => _.includes(questions, value, Number(index) + 1))
```

Comment:
a slightly more efficient version of this, sort the array then loop and for each element check if next element is a duplicate.